tests/compiler/std/typedarray.optimized.wat

```diff
@@ -32570,7 +32562,426 @@
    end
   end
  )
- (func $start:std/typedarray (; 520 ;)
+ (func $~lib/util/sort/insertionSort<i32> (; 520 ;) (param $0 i32) (param $1 i32) (param $2 i32)
+  (local $3 i32)
+  (local $4 i32)
+  (local $5 i32)
+  (local $6 i32)
+  (local $7 i32)
+  loop $loop|0
+   block $break|0
+    local.get $4
+    local.get $1
+    i32.ge_s
+    br_if $break|0
+    local.get $0
+    local.get $4
+    i32.const 2
+    i32.shl
+    i32.add
+    i32.load
+    local.set $5
+    local.get $4
+    i32.const 1
+    i32.sub
+    local.set $3
+    loop $continue|1
+     block $break|1
+      local.get $3
+      i32.const 0
+      i32.lt_s
+      br_if $break|1
+      local.get $0
+      local.get $3
+      i32.const 2
+      i32.shl
+      i32.add
+      i32.load
+      local.set $6
+      i32.const 2
+      global.set $~lib/argc
```

Comment:
@dcodeIO could we avoid this somehow?

Reply:
That's there because we can't tell whether we are calling a trampoline under the hood, so if we do, it needs to know how many args to fill. For instance:

```ts
function a(a: i32): i32 { return a; }
function b(a: i32, b?: i32 = 1) { return a + b; }
var c = someCond ? a : b;
c(2); // must indicate one arg is present so b|trampoline inserts b=1
```